codegen/src/main/java/software/amazon/awssdk/codegen/poet/builder/BaseClientBuilderClass.java

```diff
@@ -257,12 +257,31 @@ private MethodSpec finalizeServiceConfigurationMethod() {
                        .addCode("    c.dualstackEnabled(config.option($T.DUALSTACK_ENDPOINT_ENABLED));", AwsClientOption.class)
                        .addCode("}");
             }
+
+            if (model.getCustomizationConfig().getServiceConfigHasFipsConfig()) {
```

Comment:
Can we add a codegen test for this? I guess we can probably test both `serviceConfigHasFipsConfig` and `serviceConfigHasDualstackConfig` together

Reply:
Done